### PR TITLE
chore/MM-37187 : Upgrade webpack and its related dependencies Part 1 - Nov 21

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Dependency updates for :

| Dependency | From | To | Upgrade type | Changelog | Notes |
|---|---|---|---|---|---|
|copy-webpack-plugin|9.0.1|10.0.0|Major|[Diff](https://github.com/webpack-contrib/copy-webpack-plugin/blob/master/CHANGELOG.md#1000-2021-11-17)| Now requires min of 12.2 Nodejs|
|html-webpack-plugin|5.3.2|5.5.0|Minor|[Diff](https://github.com/jantimon/html-webpack-plugin/blob/main/CHANGELOG.md#550-2021-10-25)|Updates html-minifier-terser to v6, adds ability to add `type=module` in script tag of html |
|image-webpack-loader|7.0.1|8.0.1|Major|[Diff](https://github.com/tcoopman/image-webpack-loader/blob/master/CHANGELOG.md#800)|Now requires min of Node 12, removes support for webpack 1,2,3|
|webpack|5.41.1|5.64.2|Minor|From 1 to [4 release page](https://github.com/webpack/webpack/releases?page=4)|No breaking changes|
|webpack-cli|4.7.2|4.9.1|Minor|[Diff](https://github.com/webpack/webpack-cli/blob/master/CHANGELOG.md#491-2021-10-18)|cli help command improved, npx webpack-cli info, bug fixes when empty config is passed etc|
|webpack-livereload-plugin|3.0.1|3.0.2|Patch|[Diff](https://github.com/statianzo/webpack-livereload-plugin/compare/v3.0.1...v3.0.2)|New `useSourceSize` for checking reload if file changes have taken place|
|webpack-dev-server|3.11.2|4.5.0|Major|[Migration](https://github.com/webpack/webpack-dev-server/blob/master/migration-v4.md)|Explanation given in below comments|

Removed stale dependency for :
- ~terser-webpack-plugin~ : Will be removed in ticket which deals with removing of storybook
- @types/webpack-env

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37187

#### Related Pull Requests
None

#### Screenshots
None

#### Release Note

```release-note

```

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -417,9 +417,7 @@ if (targetIsDevServer) {
         devtool: 'eval-cheap-module-source-map',
         devServer: {
             hot: true,
-            injectHot: true,
```
Comment (M-ZubairAhmed) [inline webpack.config.js:420]: injectHot is now removed, sets automatically when `hot: true`

File: webpack.config.js [hunk 2]
```diff
@@ -442,8 +440,9 @@ if (targetIsDevServer) {
                 ws: true,
             }],
             port: 9005,
-            watchContentBase: true,
```
Comment (M-ZubairAhmed) [inline webpack.config.js:445]: `watchContentBase` was removed in favor of static options

**Review by hmhealey — APPROVED** [review]

File: webpack.config.js [hunk 3]
```diff
@@ -442,8 +440,9 @@ if (targetIsDevServer) {
...
-            writeToDisk: false,
+            devMiddleware: {
+                writeToDisk: true,
```
Comment (jespino) [inline webpack.config.js:444]: Any explanation for this change?

Comment (M-ZubairAhmed) [reply webpack.config.js:444]: Oops its a mistake, reverting back

**Review by jespino — APPROVED** [review]
LGTM

**Review by calebroseland — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thank you  @M-ZubairAhmed 
Here's the E2E report https://docs.google.com/document/d/19Xkh0ykk-eNhqbqcX8otVQI-am43fXWDoyhCHjNgq4M/edit
There are some invite test failing that can be disregarded. The fix for these was just merged so I may re-run the E2Es. 
Thera are few that are failing (not related to invite) please take a look and see if they are related to dependency update. Thanks
- Slack parsing message 
- MM-T951 Reopened DM shows archived
- MM-T598 Search filter Dates work without leading zeros

**Review by jgilliam17 — APPROVED** [review]
Thanks @M-ZubairAhmed 
Tested, looks good to merge.
- Verified [E2E report (recent re-run)](https://automation-dashboard.vercel.app/cycles/5a3a99a9-608b-4cb4-80b1-414ba4eff1ef) no PR related failures.

**Pull request merged**